executor/sort.go

```diff
@@ -108,7 +209,26 @@ func (e *SortExec) fetchRowChunks(ctx context.Context) error {
 		if rowCount == 0 {
 			break
 		}
-		e.rowChunks.Add(chk)
+		if e.alreadySpilled() {
+			// append chk to disk.
+			err := e.rowChunksInDisk.Add(chk)
+			if err != nil {
+				return err
+			}
+		} else {
+			e.rowChunks.Add(chk)
+			// In this pr, it must use external sorting
+			// the check should be modified before we merge this pr.
+			if atomic.LoadUint32(&e.exceeded) == 0 {
+				e.rowChunksInDisk, err = e.spillToDisk()
+				if err != nil {
+					return err
+				}
+				e.rowChunks = nil // GC its internal chunks.
+				e.memTracker.Consume(-e.memTracker.BytesConsumed())
```

Comment:
Seem not quite right here, we should only release the memory usage of `rowChunks`, otherwise it may release extra memory consumed in `initPointers`.